lib/wasm-interface/README.md

```diff
@@ -0,0 +1,88 @@
+  (func (import "wasi_unstable" "fd_fdstat_set_flags") (param i32 i32) (result i32))
+  (func (import "wasi_unstable" "fd_fdstat_set_rights") (param i32 i64 i64) (result i32))
+  (func (import "wasi_unstable" "fd_filestat_get") (param i32 i32) (result i32))
+  (func (import "wasi_unstable" "fd_filestat_set_size") (param i32 i64) (result i32))
+  (func (import "wasi_unstable" "fd_filestat_set_times") (param i32 i64 i64 i32) (result i32))
+  (func (import "wasi_unstable" "fd_pread") (param i32 i32 i32 i64 i32) (result i32))
+  (func (import "wasi_unstable" "fd_prestat_get") (param i32 i32) (result i32))
+  (func (import "wasi_unstable" "fd_prestat_dir_name") (param i32 i32 i32) (result i32))
+  (func (import "wasi_unstable" "fd_pwrite") (param i32 i32 i32 i64 i32) (result i32))
+  (func (import "wasi_unstable" "fd_read") (param i32 i32 i32 i32) (result i32))
+  (func (import "wasi_unstable" "fd_readdir") (param i32 i32 i32 i64 i32) (result i32))
+  (func (import "wasi_unstable" "fd_renumber") (param i32 i32) (result i32))
+  (func (import "wasi_unstable" "fd_seek") (param i32 i64 i32 i32) (result i32))
+  (func (import "wasi_unstable" "fd_sync") (param i32) (result i32))
+  (func (import "wasi_unstable" "fd_tell") (param i32 i32) (result i32))
+  (func (import "wasi_unstable" "fd_write") (param i32 i32 i32 i32) (result i32))
+  (func (import "wasi_unstable" "path_create_directory") (param i32 i32 i32) (result i32))
+  (func (import "wasi_unstable" "path_filestat_get") (param i32 i32 i32 i32 i32) (result i32))
+  (func (import "wasi_unstable" "path_filestat_set_times") (param i32 i32 i32 i32 i64 i64 i32) (result i32))
+  (func (import "wasi_unstable" "path_link") (param i32 i32 i32 i32 i32 i32 i32) (result i32))
+  (func (import "wasi_unstable" "path_open") (param i32 i32 i32 i32 i32 i64 i64 i32 i32) (result i32))
+  (func (import "wasi_unstable" "path_readlink") (param i32 i32 i32 i32 i32 i32) (result i32))
+  (func (import "wasi_unstable" "path_remove_directory") (param i32 i32 i32) (result i32))
+  (func (import "wasi_unstable" "path_rename") (param i32 i32 i32 i32 i32 i32) (result i32))
+  (func (import "wasi_unstable" "path_symlink") (param i32 i32 i32 i32 i32) (result i32))
+  (func (import "wasi_unstable" "path_unlink_file") (param i32 i32 i32) (result i32))
+  (func (import "wasi_unstable" "poll_oneoff") (param i32 i32 i32 i32) (result i32))
+  (func (import "wasi_unstable" "proc_exit") (param i32))
+  (func (import "wasi_unstable" "proc_raise") (param i32) (result i32))
+  (func (import "wasi_unstable" "random_get") (param i32 i32) (result i32))
+  (func (import "wasi_unstable" "sched_yield") (result i32))
+  (func (import "wasi_unstable" "sock_recv") (param i32 i32 i32 i32 i32 i32) (result i32))
+  (func (import "wasi_unstable" "sock_send") (param i32 i32 i32 i32 i32) (result i32))
+  (func (import "wasi_unstable" "sock_shutdown") (param i32 i32) (result i32))
+)
+```
```

Comment:
I'm not sure but for WASIX we'd probably have to do a rework, for now I just 1:1 copied the necessary code from wapm-cli.

Reply:
I don't know the context here, but: for wasi there is also the `snapshot1` , which is also commonly used , not just `wasi_unstable`.

And yes, wasix will have a different namespace and additional functions.
Also, there is a 32bit and a 64bit version.
(Wasmer doesn't support 64bit yet though)

Reply:
For now I'll leave this as-is since I'm only porting the functionality we already have in wapm. If we notice problems after 3.2, we can still go in and edit the definitions again.